net/Socket.cpp

```diff
@@ -1251,11 +1309,27 @@ bool Socket::isLocal() const
 std::shared_ptr<Socket> LocalServerSocket::accept()
 {
     const int rc = ::accept4(getFD(), nullptr, nullptr, SOCK_NONBLOCK | SOCK_CLOEXEC);
+    if (rc < 0)
+    {
+        constexpr const char * messagePrefix = "Failed to accept. (errno: ";
+        const int cause = errno;
+        if (isRetryAcceptError(cause))
+        {
+            // Recoverable error, ignore to retry
+            LOG_DBG(messagePrefix << std::to_string(cause) << ", " << std::strerror(cause) << ')');
+        }
+        else
+        {
+            // Unrecoverable error, throw, which will close `ServerSocket` and discard it from `AcceptPoll` during `SocketPoll::poll`
```

Comment:
The comment here claims it throws, but it actually exits now

Reply:
good catch, thx. yup, we moved to fatal-exit. throwing only inside the unit-test hooks injecting failures (otherwise test can't pass).